src/main/kotlin/pt/up/fe/ni/website/backend/controller/RoleController.kt

```diff
@@ -0,0 +1,90 @@
+package pt.up.fe.ni.website.backend.controller
+
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import pt.up.fe.ni.website.backend.dto.entity.RoleDto
+import pt.up.fe.ni.website.backend.dto.permissions.PermissionsDto
+import pt.up.fe.ni.website.backend.model.permissions.Permissions
+import pt.up.fe.ni.website.backend.service.RoleService
+
+@RestController
+@RequestMapping("/roles")
+class RoleController(private val roleService: RoleService) {
+    @GetMapping
+    fun getAllRoles() = roleService.getAllRoles()
+
+    @GetMapping("/{id}")
+    fun getRole(@PathVariable id: Long) = roleService.getRole(id)
+
+    @PostMapping("/new")
+    fun createNewRole(@RequestBody dto: RoleDto) = roleService.createNewRole(dto)
+
+    @DeleteMapping("/{id}")
+    fun deleteRole(@PathVariable id: Long): Map<String, String> {
+        roleService.deleteRole(id)
+        return emptyMap()
+    }
+
+    @PostMapping("/{id}/grant")
+    fun grantPermissionRole(
+        @PathVariable id: Long,
+        @RequestBody permissionsDto: PermissionsDto
+    ): Map<String, String> {
+        roleService.grantPermissionToRole(id, Permissions(permissionsDto.permissions))
```

Comment:
By serializing Permissions directly, the request body becomes `[1,2,3]`, for example, and with the DTO it becomes `{permissions:[1,2,3]}`which in my opinion, is more readable and more consistent with other API calls, wdyt? 

Maybe we need a solution for one parameter solutions like this, and userId, because creating an DTO for every one parameter body is not very sustainable.

Reply:
I see and agree with you. I'm not aware of a better solution, I think DTOs are the standard go-to in cases like this one